core/src/main/java/hudson/cli/BuildCommand.java

```diff
@@ -158,7 +158,7 @@ protected int run() throws Exception {
 
         if (!job.isBuildable()) {
             String msg = Messages.BuildCommand_CLICause_CannotBuildUnknownReasons(job.getFullDisplayName());
-            if (job instanceof AbstractProject<?, ?> && ((AbstractProject<?, ?>)job).isDisabled()) {
+            if (job instanceof ParameterizedJobMixIn.ParameterizedJob && ((ParameterizedJobMixIn.ParameterizedJob) job).isDisabled()) {
```

Comment:
It is indeed a little tricky. `Job` does define `buildable` (without any explanation), but the idea here was to define `disabled` in `ParameterizedJob` since that is now the home for “`Job` which we could potentially request a build of, perhaps with parameters, so long as that is not disabled” (i.e., basically everything except `ExternalJob`).

`BuildableItem` (not to be confused with `Queue.BuildableItem`!), well, this interface is a historical relic: very little code actually deals with objects declared to be of this type and expects `scheduleBuild` overloads to work. I am implementing it for completeness; `AbstractProject` and `WorkflowJob`, the only `ParameterizedJob`s out there, already implemented it anyway, so this is just to make it easier to provide default implementations for those four methods (mostly superseded by `scheduleBuild2` overloads).